components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -58,26 +76,40 @@ class FxaPushSupportFeature(
 internal class AccountObserver(
     private val context: Context,
     private val accountManager: FxaAccountManager,
-    private val observer: DeviceConstellationObserver,
+    private val push: AutoPushFeature,
     private val lifecycleOwner: LifecycleOwner,
     private val autoPause: Boolean
 ) : SyncAccountObserver {
+
     private val logger = Logger("AccountObserver")
+    private val constellationObserver = ConstellationObserver(context, push)
 
     override fun onAuthenticated(account: OAuthAccount, authType: AuthType) {
+        // We need a new subscription only when we have a new account.
+        // This is removed when an account logs out.
+        if (authType != AuthType.Existing) {
+            logger.debug("Subscribing for ${PushType.Services} events.")
+
+            push.subscribeForType(PushType.Services)
+        }
+
         accountManager.withConstellation { constellation ->
-            constellation.registerDeviceObserver(observer, lifecycleOwner, autoPause)
+            constellation.registerDeviceObserver(constellationObserver, lifecycleOwner, autoPause)
         }
     }
 
     override fun onLoggedOut() {
+        logger.debug("Un-subscribing for ${PushType.Services} events.")
+
+        push.unsubscribeForType(PushType.Services)
+
         // Delete renewal pref.
```

Comment:
Can you change this comment to say why you're deleting it? Otherwise, it's not helpful and it's not at all obvious what this thing is.